### DISCUSSION TITLE
Can i log couchdb logs into Database or any other place of my choice

### DISCUSSION THREAD

**Discussion by ghnash** [apache/couchdb#3316] [Q&A] [answered]
Hello CouchDB Community,

We know the location of couchdb logs and it writes into this file /var/log/couchdb/couch.log. 

My Question is can i write the log into other place of our choice like in Database or someother file. If yes how? Anyone has any idea? 
OR couchdb provided some logs API which i can access and write the logs into my Database or file of my choice. Please suggest.

any idea/response will be highly appreciated. 

Happy Holidays and Happy New Year!!!

Thanks
Nash

**Comment by jjrodrig** [comment, marked as answer]
You can check the logging options here https://docs.couchdb.org/en/stable/config/logging.html

CouchDB supports the following log writers:
- stderr: Logs are sent to stderr.
- file: Logs are sent to the file set in log file.
- syslog: Logs are sent to the syslog daemon.
- journald: Logs are sent to stderr without timestamp and log levels compatible with sd-daemon.

You can change the log file by setting the **file** param in your local config file.
```
[log]
file = /var/log/couchdb/couch.log
```

**Reply by ghnash** [reply]
Thanks @jjrodrig for config option. 

can i write the log into DB if yes what should be my log writer setting? Thanks and appreciate your help :)

**Reply by jjrodrig** [reply]
This is not directly supported by CouchDB. You may explore some specific tools for logging collection that could process the log files or the syslog events and redirect the log to some other storage. There are many tools that can help on this: fluentd, logstash, Apache Flume, rsyslog, ...

**Reply by ghnash** [reply]
Thanks @jjrodrig. I know about logstash for ElasticSearch. But I was looking for custom solution, basically i wanted to write my own custom API and it should keep polling at certian interval and pull the data from logs and push into the data storage which i have.

I appreciate your reply and help. Thank you so much 👍  :)
